--- start-msvc11-x64.bat.orig	2012-07-18 19:44:24 +0200
+++ start-msvc11-x64.bat	2012-09-30 17:48:59 +0200
@@ -30,7 +30,7 @@
     )
 
     rem Prepend MSVC paths
-    call "%VC11EXPRESSDIR%\bin\amd64\vcvars64.bat" 2>nul
+    call "%VC11EXPRESSDIR%\bin\x86_amd64\vcvarsx86_amd64.bat" 2>nul
     if "%DevEnvDir%"=="" (
       rem Might be using a compiler that shipped with an SDK, so manually set paths
       SET "PATH=%VC11EXPRESSDIR%\Bin\amd64;%VC11EXPRESSDIR%\Bin;%PATH%"
@@ -58,7 +58,7 @@
     rem Prepend MSVC paths
     rem The Win7 SDK (or newer) should automatically integrate itself into vcvars32.bat
     ECHO Using VC 2012 built-in SDK
-    call "%VC11DIR%\bin\amd64\vcvars64.bat"
+    call "%VC11DIR%\bin\x86_amd64\vcvarsx86_amd64.bat"
 )
 
 if "%VC11DIR%"=="" (
