diff -ruN --strip-trailing-cr gettext-runtime-0.18.orig/gettext-runtime/intl/intl.def gettext-runtime-0.18/gettext-runtime/intl/intl.def
--- gettext-runtime-0.18.orig/gettext-runtime/intl/intl.def	2012-09-28 21:56:54 +0200
+++ gettext-runtime-0.18/gettext-runtime/intl/intl.def	2012-09-28 21:56:44 +0200
@@ -1,4 +1,3 @@
-LIBRARY libintl.DLL
 EXPORTS
 ;_imp___ctype_ptr
 _nl_expand_alias
@@ -43,6 +42,11 @@
 libintl_nl_default_default_domain
 libintl_nl_default_dirname
 libintl_nl_domain_bindings
+libintl_vfprintf
+libintl_vprintf
+libintl_vsprintf
+libintl_printf
+libintl_sprintf
 ;libintl_relocate
 ;libintl_set_relocation_prefix
 libintl_setlocale
